src/location.c

```diff
@@ -530,24 +531,21 @@ handle_start_in_thread_func (GTask *task,
       g_variant_builder_add (&access_opt_builder, "{sv}",
                              "icon", g_variant_new_string ("find-location-symbolic"));
 
-      if (g_str_equal (app_id, ""))
+      info = xdp_app_info_load_app_info (request->app_info);
+      if (info == NULL)
         {
           title = g_strdup (_("Grant Access to Your Location?"));
           subtitle = g_strdup (_("An application wants to use your location."));
         }
       else
         {
-          g_autofree char *id = NULL;
-          g_autoptr(GDesktopAppInfo) info = NULL;
           const char *name;
 
-          id = g_strconcat (app_id, ".desktop", NULL);
-          info = g_desktop_app_info_new (id);
-          name = g_app_info_get_display_name (G_APP_INFO (info));
-
+          name = g_app_info_get_display_name (info);
           title = g_strdup_printf (_("Give %s Access to Your Location?"), name);
-          if (g_desktop_app_info_has_key (info, "X-Geoclue-Reason"))
-            subtitle = g_desktop_app_info_get_string (info, "X-Geoclue-Reason");
+          if (G_DESKTOP_APP_INFO (info) &&
```

Comment:
This cast and NULL check seems unnecessary.

Reply:
it was meant to be a an `IS` check, but however it's always true in the current code but `GAppInfo` may also be implemented differently.